src/main/java/org/sagebionetworks/web/client/security/AuthenticationControllerImpl.java

```diff
@@ -202,8 +202,10 @@ public void initializeFromExistingAccessTokenCookie(
       new FutureCallback<String>() {
         @Override
         public void onSuccess(String accessToken) {
-          currentUserAccessToken = accessToken;
-          clearQueryClientCache();
+          if (!Objects.equals(currentUserAccessToken, accessToken)) {
+            resetQueryClientCache();
+            currentUserAccessToken = accessToken;
+          }
```

Comment:
This runs on a timer, so every minute we were clearing the cache!

Reply:
I understand that the check should be executed every minute, but the access token should not be changing.  Right?

Reply:
Correct. Now that we have this condition, the cache will only be reset if the token has changed.